java/bundles/org.eclipse.set.utils.table/src/org/eclipse/set/utils/table/sorting/CompareRouteAndKmCriterion.java

```diff
@@ -33,10 +41,15 @@
  */
 public class CompareRouteAndKmCriterion
 		extends AbstractCompareWithDependencyOnServiceCriterion<TableRow> {
-
+	private final Logger logger = LoggerFactory
+			.getLogger(CompareRouteAndKmCriterion.class);
 	private final SortDirectionEnum direction;
 	private final Function<Ur_Objekt, Punkt_Objekt> getPunktObjectFunc;
 	private final NumericCellComparator numericComparator;
+	private static final String KILOMETRIERUNG_PATTERN = "(?<numberPrefix>-)?(?<numberN>[1-9]\\d{0,2}|0),((?<numberD1>\\d{3})|(?<numberD2>\\d)(?<numberN2Prefix>[\\+\\-])(?<numberN2>[1-9]\\d{0,4}))"; //$NON-NLS-1$
+	private static final String EXTRA_LENGTH_GROUP_NAME = "numberN2"; //$NON-NLS-1$
+	private static final String EXTRA_LENGTH_PREFIX = "numberN2Prefix"; //$NON-NLS-1$
```

Comment:
@TruongQuangSB Can't we just the following PATTERN

```suggestion
	private static final String KILOMETRIERUNG_PATTERN = "(?<numberPrefix>-)?(?<numberN>[1-9]\\d{0,2}|0),((?<numberD1>\\d{3})|(?<numberD2>\\d)(?<numberN2>[\\+\\-][1-9]\\d{0,4}))"; //$NON-NLS-1$
	private static final String EXTRA_LENGTH_GROUP_NAME = "numberN2"; //$NON-NLS-1$
	private static final String EXTRA_LENGTH_PREFIX = "numberN2Prefix"; //$NON-NLS-1$
```

This would also make the handling `convertToDouble` more easier in my opinion